discovery/src/server.rs

```diff
@@ -55,21 +55,19 @@ impl RequestHandler {
 
         let discovery = Self {
             config,
-            username: None,
+            username: Mutex::new(None),
             keys: DhLocalKeys::random(&mut rand::thread_rng()),
             tx,
         };
 
         (discovery, rx)
     }
 
-    fn handle_get_info(&self) -> Response<hyper::Body> {
+    async fn handle_get_info(&self) -> Response<hyper::Body> {
         let public_key = BASE64.encode(self.keys.public_key());
         let device_type: &str = self.config.device_type.into();
-        let mut active_user = String::new();
-        if let Some(username) = &self.username {
-            active_user = username.to_string();
-        }
+        let maybe_username = self.username.lock().await;
+        let active_user = maybe_username.as_deref().unwrap_or("");
```

Comment:
Can you comment why the `unwrap_or` is "safe" here? Should it always succeed?

Reply:
The behavior is the same as previously, though the fixup modifies this hunk. There is no panic risk.